src/com/oltpbenchmark/schema/defaultschema/DefaultSchemaManager.java

```diff
@@ -0,0 +1,121 @@
+            db_connection.commit();
+        }
+    }
+
+    public void createIndexes() throws SQLException {
+        execute("CREATE INDEX idx_customer_name ON customer ((c_w_id,c_d_id) HASH,c_last,c_first)");
+        execute("CREATE UNIQUE INDEX idx_order ON oorder ((o_w_id,o_d_id) HASH,o_c_id,o_id DESC)");
+    }
+    
+    @Override
+    public void enableForeignKeyConstraints() throws SQLException {
+        execute("ALTER TABLE DISTRICT DROP CONSTRAINT IF EXISTS D_FKEY_W");
+        execute("ALTER TABLE DISTRICT ADD CONSTRAINT D_FKEY_W FOREIGN KEY (D_W_ID) " +
+                "REFERENCES WAREHOUSE(W_ID) NOT VALID");
+
+        execute("ALTER TABLE CUSTOMER DROP CONSTRAINT IF EXISTS C_FKEY_D");
+        execute("ALTER TABLE CUSTOMER ADD CONSTRAINT C_FKEY_D FOREIGN KEY (C_W_ID, C_D_ID) " +
+                "REFERENCES DISTRICT (D_W_ID, D_ID) NOT VALID");
+
+        execute("ALTER TABLE STOCK DROP CONSTRAINT IF EXISTS S_FKEY_W");
+        execute("ALTER TABLE STOCK DROP CONSTRAINT IF EXISTS S_FKEY_I");
+        execute("ALTER TABLE STOCK ADD CONSTRAINT S_FKEY_W FOREIGN KEY(S_W_ID) " +
+                "REFERENCES WAREHOUSE(W_ID) NOT VALID");
+        execute("ALTER TABLE STOCK ADD CONSTRAINT S_FKEY_I FOREIGN KEY(S_I_ID)  " +
+                "REFERENCES ITEM(I_ID) NOT VALID");
+
+        execute("ALTER TABLE OORDER DROP CONSTRAINT IF EXISTS O_FKEY_C");
+        execute("ALTER TABLE OORDER ADD CONSTRAINT O_FKEY_C FOREIGN KEY (O_W_ID, O_D_ID, O_C_ID) " +
+                "REFERENCES CUSTOMER (C_W_ID, C_D_ID, C_ID) NOT VALID");
+
+        execute("ALTER TABLE NEW_ORDER DROP CONSTRAINT IF EXISTS NO_FKEY_O");
+        execute("ALTER TABLE NEW_ORDER ADD CONSTRAINT NO_FKEY_O FOREIGN KEY " +
+                "(NO_W_ID, NO_D_ID, NO_O_ID) REFERENCES OORDER (O_W_ID, O_D_ID, O_ID) NOT VALID");
+
+        execute("ALTER TABLE HISTORY DROP CONSTRAINT IF EXISTS H_FKEY_C");
+        execute("ALTER TABLE HISTORY DROP CONSTRAINT IF EXISTS H_FKEY_D");
+        execute("ALTER TABLE HISTORY ADD CONSTRAINT H_FKEY_C FOREIGN KEY " +
+                "(H_C_W_ID, H_C_D_ID, H_C_ID) REFERENCES CUSTOMER (C_W_ID, C_D_ID, C_ID) NOT VALID");
+        execute("ALTER TABLE HISTORY ADD CONSTRAINT H_FKEY_D FOREIGN KEY " +
+                "(H_W_ID, H_D_ID) REFERENCES DISTRICT (D_W_ID, D_ID) NOT VALID");
+
+        execute("ALTER TABLE ORDER_LINE DROP CONSTRAINT IF EXISTS OL_FKEY_O");
+        execute("ALTER TABLE ORDER_LINE DROP CONSTRAINT IF EXISTS OL_FKEY_S");
+        execute("ALTER TABLE ORDER_LINE ADD CONSTRAINT OL_FKEY_O FOREIGN KEY " +
+                "(OL_W_ID, OL_D_ID, OL_O_ID) REFERENCES OORDER (O_W_ID, O_D_ID, O_ID) NOT VALID");
+        execute("ALTER TABLE ORDER_LINE ADD CONSTRAINT OL_FKEY_S FOREIGN KEY " +
+                "(OL_SUPPLY_W_ID, OL_I_ID) REFERENCES STOCK (S_W_ID, S_I_ID) NOT VALID");
+    }
+    
+    @Override
+    public void createSqlProcedures() throws Exception {
+        Statement st = db_connection.createStatement();
```

Comment:
Statement should be closed after the usage to release resources

```
try (Statement stmt = db_connection.createStatement()) {
```

Reply:
Good catch. Fixed